cmake/helpers.cmake

```diff
@@ -91,18 +91,24 @@ function(add_umf_target_compile_options name)
         target_compile_options(
             ${name}
             PRIVATE -fPIC
+                    # -fPIE
                     -Wall
+                    -Wextra
+                    -Werror
                     -Wpedantic
                     -Wempty-body
                     -Wunused-parameter
+                    -Wformat
+                    -Wformat-security
+                    -fstack-protector-strong
```

Comment:
I think this one might make performance suffer as it adds additional checks on a program run. Should it be included only in the Debug builds or left in the DEVELOPER_MODE flag? Same with the FORTIFY_SOURCE.

Reply:
moved